src/main/java/me/moodcat/api/UserAPI.java

```diff
@@ -0,0 +1,103 @@
+package me.moodcat.api;
+
+import java.util.List;
+
+import javax.ws.rs.DefaultValue;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.MediaType;
+
+import me.moodcat.database.controllers.UserDAO;
+import me.moodcat.database.entities.User;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.inject.Inject;
+import com.google.inject.persist.Transactional;
+
+/**
+ * The API for the room.
+ */
+@Path("/api/users/")
+@Produces(MediaType.APPLICATION_JSON)
+public class UserAPI {
+
+    /**
+     * Access to the user DAO.
+     */
+    private final UserDAO userDAO;
+
+    @Inject
+    @VisibleForTesting
+    public UserAPI(final UserDAO userDAO) {
+        this.userDAO = userDAO;
+    }
+
+    @GET
+    @Transactional
+    public List<User> getUsers() {
+        return userDAO.getAll();
+    }
+
+    /**
+     * Get the user according to the provided id.
+     *
+     * @param id
+     *            The id of the user to find.
+     * @return The user according to the id.
+     * @throws IllegalArgumentException
+     *             If the id is null.
+     */
+    @GET
+    @Path("{id}")
+    @Transactional
+    public User getUser(@PathParam("id") final int userId) {
+        return userDAO.retrieveBySoundcloudId(userId);
+    }
+
+    /**
+     * Returns the current user
+     * 
+     * @return
+     */
+    @GET
+    @Path("me")
+    @Transactional
+    public User getMe() {
+        // FIXME: Hardcoded until Oauth works
```

Comment:
After merge to the master this can be refactored.

* Inject a `@Named("current.user") Provider<User> currentUserProvider` to the resource. Be sure that this annotation is at the constructor parameter.
* Call `currentUserProvider.get()` to get the current user
* The soundcloud OAuth token should be sent as param to the request in order to work, otherwise an 403 FORBIDDEN will be thrown. https://github.com/MoodCat/MoodCat.me-Frontend/blob/master/app/scripts/controllers/main.js#L109 .